### PR TITLE
Fix Q.rational(x) for floating point x

### LINKED ISSUE #12168: Wrong answer for ask(Q.rational(x)) for a float x
Wrong answer for `ask(Q.rational(x))` for `x` a float instance
e.g
```
>>> ask(Q.rational(math.pi))
True  # this should return None
```
I want to solve this Issue can someone guide me.

### PR DESCRIPTION
**Pull request by meet7202**
Fixes #12168

### CODE REVIEW

File: sympy/assumptions/handlers/sets.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def Pow(expr, assumptions):
                 return False
 
     Rational, Float = \
-        [staticmethod(CommonHandler.AlwaysTrue)]*2 # Float is finite-precision
+        [staticmethod(CommonHandler.AlwaysFalse)]*2 # Float is finite-precision
```
Comment (jksuom) [inline sympy/assumptions/handlers/sets.py]: This is not the correct fix. There are two handlers on this line. They must be separated. The one for `Rational` should be left as it is. And a new handler (returning `None`) should be defined for `Float`.

**Comment by meet7202** [comment]
1
```
Rational = \
        [staticmethod(CommonHandler.AlwaysTrue)]*2 # Float is finite-precision
    Float = \
        [staticmethod(CommonHandler.AlwaysFalse)]*2 # Float is finite-precision
    
```
or
2 
```
Rational = \
        [staticmethod(CommonHandler.AlwaysTrue)]*2 # Float is finite-precision
    Float = \
        [staticmethod(CommonHandler.AlwaysNone)]*2 # Float is finite-precision
    
```
with
```

  @staticmethod
    def AlwaysNone(expr, assumptions):
        return None
``` 
in class CommonHandler ?
I think second one looks good , but It wouldn't worked for me when I tested that's why I am asking you @jksuom .

**Comment by jksuom** [comment]
```
    Rational = \
        [staticmethod(CommonHandler.AlwaysTrue)]*2 # Float is finite-precision
```
Rational should not be a list but a method. 
```
     Float = \
        [staticmethod(CommonHandler.AlwaysNone)]*2 # Float is finite-precision
```
It is probably overkill to define an AlwaysNone handler as it seems not to be used elsewhere. It would suffice to write an ordinary staticmethod `def Float` returning `None`. 

The comment about finite precision is irrelevant in either case.

Some tests should be fixed to satisfy the changed results. You can find them locally by running bin/test query and bin/doctest.

**Comment by meet7202** [comment]
@jksuom , But If Float will return None in all cases then even simple cases like 
```
>>>ask(Q.rational(124.2))
#No Output as Return None
```
won't get output, Is there any way to Identify `math.pi` before it will be converted to `float` so that it can be added to list with exception cases with pi,Exp1, GoldenRatio etc. ?

**Comment by jksuom** [comment]
> Is there any way to Identify math.pi before it will be converted to float

It seems that this takes place in the interpreter. (In CPython code, a double is initialized with an approximation of pi, and this is then converted to a Python float.)

**Comment by gxyd** [comment]
Closing this, since issue has been fixed by #13870.

**Pull request closed without merging**